File: Frontend/src/app/Services/show.service.ts
import { Injectable } from '@angular/core';
import {HttpClient, HttpHeaders} from "@angular/common/http";
import {Show} from "../Models/show.model";

@Injectable({
  providedIn: 'root'
})
export class ShowService {

  httpOptions = {
    headers :new HttpHeaders({
      'Content-Type':'application/json'
    })
  }

  constructor(private http: HttpClient) { }

  addShow(show: any) {
    return this.http.post("http://localhost:8090/shows/add", show, this.httpOptions);
  }

  getAllShows() {
    return this.http.get<Show[]>("http://localhost:8090/shows/getAll");
  }

  getShowById(theatreId: any, movieId: any, locationId: any) {
    return this.http.get<Show[]>(`http://localhost:8090/shows/getbyid/${movieId}/${locationId}/${theatreId}`);
  }
}

File: Backend/src/main/java/com/example/backend/Service/Show/ShowService.java
package com.example.backend.Service.Show;

import com.example.backend.Model.Booking;
import com.example.backend.Model.Shows;

import java.util.List;

public interface ShowService {
    public List<Shows> listAllShows();
    public boolean addShow(Shows show);
    public List<Shows> showsById(int movieId, int locationId, int TheatreId);
  //  public void updateShow(Booking booking);
   // public void rollbackShow(String showid);]
    public List<Shows> showsByAllIds(int movieId, int locationId, int TheatreId,int showid);
}
